src/infra/dex/oneinch/dto.rs

```diff
@@ -105,6 +108,7 @@ impl Query {
             amount: order.amount.get(),
             slippage: Slippage::from_domain(slippage),
             gas_price: Some(gas_price.0 .0),
+            origin: order.owner,
```

Comment:
The order owner is pretty much irrelevant in this context.
At the point interactions generated by `/swap` get executed the settlement contract already took the `sell_tokens` from the user so it would be `msg.sender`.
BUT in their docs they emphasize that `origin` has to be an `EOA` which makes me think they actually want the `tx.origin` here which would actually be the address of the oneinch solver.

Reply:
A bit more context: https://cowservices.slack.com/archives/C0371SB243E/p1718722940784089
They requested the end user address.

Reply:
Okay, so this is mostly for regulatory purposes but from my understanding this should not really change anything.
Either the address is not on a block list then they will return the same calldata as before or the address is on some list and it should return an error.
But in any case this seems like a reasonable ask but I would then document the `origin` field differently because that their docs don't apply to use for at least 2 reasons:
1. the owner might not be an EOA but a smart contract
2. the owner is neither `tx.origin` nor `msg.sender`

Reply:
Thanks, makes sense. Changed the doc.